src/plugins/kibana_usage_collection/server/collectors/csp/csp_collector.ts

```diff
@@ -60,8 +56,11 @@ export function createCspCollector(server: Server): CollectorOptions<Usage> {
   };
 }
 
-export function registerCspCollector(usageCollection: UsageCollectionSetup, server: Server): void {
-  const collectorConfig = createCspCollector(server);
-  const collector = usageCollection.makeUsageCollector<Usage>(collectorConfig);
+export function registerCspCollector(
+  usageCollection: UsageCollectionSetup,
+  http: HttpServiceSetup
+): void {
+  const collectorOptions = createCspCollector(http);
+  const collector = usageCollection.makeUsageCollector<Usage>(collectorOptions);
   usageCollection.registerCollector(collector);
```

Comment:
The `scripts/telemetry_check.js` was kinda picky on the syntax I had to use here: I was not allowed to directly call `usageCollection.makeUsageCollector(createCspCollector(http))` and I was forced to manually specify the `<Usage>` type that is already implicitly defined when calling `makeUsageCollector`